crates/networking/discv5/src/eth2.rs

```diff
@@ -0,0 +1,103 @@
+use alloy_primitives::{B256, aliases::B32, fixed_bytes};
+use ethereum_ssz::Encode;
+use ethereum_ssz_derive::{Decode, Encode};
+use ream_consensus::fork_data::ForkData;
+
+// Constants
+pub const FAR_FUTURE_EPOCH: u64 = 18446744073709551615;
+pub const GENESIS_FORK_VERSION: B32 = fixed_bytes!("0x00000000");
+pub const GENESIS_VALIDATORS_ROOT: B256 =
+    fixed_bytes!("0x0000000000000000000000000000000000000000000000000000000000000000");
+pub const ENR_ETH2_KEY: &str = "eth2";
+// Types
+#[derive(Default, Debug, Encode, Decode)]
+pub struct ENRForkID {
+    pub fork_digest: B32,
+    pub next_fork_version: B32,
+    pub next_fork_epoch: u64,
+}
+
+impl ENRForkID {
+    pub fn new(current_fork_version: B32, next_fork_version: B32, next_fork_epoch: u64) -> Self {
+        let fork_digest = ForkData {
+            current_version: current_fork_version,
+            genesis_validators_root: GENESIS_VALIDATORS_ROOT,
+        }
+        .compute_fork_digest();
+
+        Self {
+            fork_digest,
+            next_fork_version,
+            next_fork_epoch,
+        }
+    }
+
+    pub fn new_pectra() -> Self {
+        // Pectra fork version and epoch values
+        let current_fork_version = B32::from_slice(&[0x03, 0x00, 0x00, 0x00]); // This should be replaced with actual Pectra fork version
+        let next_fork_version = B32::from_slice(&[0x00, 0x00, 0x00, 0x00]); // FAR_FUTURE_EPOCH version
+        let next_fork_epoch = FAR_FUTURE_EPOCH;
+
+        Self::new(current_fork_version, next_fork_version, next_fork_epoch)
+    }
+
+    pub fn as_ssz_bytes(&self) -> Vec<u8> {
+        let mut bytes = Vec::new();
+        Encode::encode(self, &mut bytes);
+        bytes
+    }
+    pub fn compute_fork_digest(current_fork_version: B32) -> B32 {
+        let fork_data = ForkData {
+            current_version: current_fork_version,
+            genesis_validators_root: GENESIS_VALIDATORS_ROOT,
+        };
+        fork_data.compute_fork_digest()
+    }
+}
+
+impl Decode for ENRForkID {
+    fn decode(bytes: &[u8]) -> Result<Self, ethereum_ssz::DecodeError> {
+        ethereum_ssz::Decode::decode(bytes)
+    }
+}
+
```

Comment:
```suggestion

impl Encodable for ENRForkID {
    fn encode(&self, out: &mut dyn bytes::BufMut) {
        let ssz_bytes = self.as_ssz_bytes();
        let bytes = Bytes::from(ssz_bytes);
        bytes.encode(out);
    }
}

impl Decodable for ENRForkID {
    fn decode(buf: &mut &[u8]) -> alloy_rlp::Result<Self> {
        let bytes = Bytes::decode(buf)?;
        let enr_fork_id = ENRForkID::from_ssz_bytes(&bytes)
            .map_err(|_| alloy_rlp::Error::Custom("Failed to decode SSZ ENRForkID"))?;
        Ok(enr_fork_id)
    }
}

```

add this code then you should be able to do

```
let enr = Enr::builder()
            .add_value(ENR_ETH2_KEY, ENRForkID::new_pectra())
            .build(&enr_local)?;
```

as doing .as_ssz_bytes() won't encode it properly